database/scrapers/event_calendar.ts

```diff
@@ -0,0 +1,214 @@
+    if (fieldIndex === -1) {
+      return null;
+    }
+    const startIndex = fieldIndex + fieldName.length + 1;
+    const endIndex = eventBlock.indexOf("\n", startIndex);
+    if (endIndex === -1) {
+      return [eventBlock.substring(startIndex).trim(), eventBlock.length];
+    }
+    return [eventBlock.substring(startIndex, endIndex).trim(), endIndex + 1];
+  }
+}
+
+export default class EventCalendarUpdater extends BaseScraperModule {
+  static name = "Event calendar";
+  static description =
+    "Update and push to the database events from the Student Council Google calendar as Google events (non-editable, removable)";
+  static taskTitle =
+    "Update event calendar with latest events from Student Council Google calendar";
+
+  private static async fetchGoogleEvents(): Promise<string> {
+    const calendarResponse = await fetch(
+      `https://calendar.google.com/calendar/ical/${CALENDAR_ID}/public/basic.ics`,
+    );
+    if (!calendarResponse.ok) {
+      throw new Error(
+        `Failed to fetch calendar events - got response status code ${calendarResponse.status}`,
+      );
+    }
+    return calendarResponse.text();
+  }
+
+  private static async getEventsInTimeframe(
+    task: TaskHandle,
+  ): Promise<GoogleCalendarEventDto[]> {
+    const calendarString = await EventCalendarUpdater.fetchGoogleEvents();
+    const parser: CalendarParser = new CalendarParser(calendarString);
+    const bound = DateTime.now().minus({ days: FETCH_LIMIT_DAYS });
+    const events: GoogleCalendarEventDto[] = [];
+    let event = parser.getLastEvent(task);
+    while (event !== null && event.startTime >= bound) {
+      events.push(event);
+      event = parser.getLastEvent(task);
+    }
+    task.update(
+      `Total of ${events.length} events found in the given timeframe (${FETCH_LIMIT_DAYS} days).`,
+    );
+    return events;
+  }
+
+  private static async upsertDatabaseEvents(task: TaskHandle) {
+    const newEvents = await EventCalendarUpdater.getEventsInTimeframe(task);
+    for (const event of newEvents) {
+      const existingEvent = await CalendarEvent.findBy(
+        "google_cal_id",
+        event.googleCalId,
+      );
+      if (existingEvent !== null) {
+        existingEvent.startTime = event.startTime;
+        existingEvent.endTime = event.endTime;
+        existingEvent.name = event.name;
+        existingEvent.location = event.location;
+        existingEvent.description = event.description;
+        await existingEvent.save();
+        task.update(`Updated event ${event.name}`);
+      } else {
+        await CalendarEvent.create({ ...event });
+        task.update(`Added new event ${event.name}`);
+      }
+    }
+  }
+
+  private static async removeOutdatedEvents() {
+    const bound = DateTime.now().minus({ days: FETCH_LIMIT_DAYS });
+    await CalendarEvent.query()
+      .where("start_time", "<", bound.toISO())
+      .whereNotNull("google_cal_id")
+      .delete();
+  }
```

Comment:
you probably should check the end time instead of start time, just in case there's some event lasting like a week or smth